File: c++/0746.cpp
// Link: https://leetcode.com/problems/min-cost-climbing-stairs/
// Runtime: 5 ms
// Memory: 14.3 MB

class Solution {
public:
  int minCostClimbingStairs(const std::vector<int> &cost) {
    std::vector<int> minCost = cost;
    for (int i = 2; i < static_cast<int>(minCost.size()); i++)
      minCost[i] += std::min(minCost[i - 1], minCost[i - 2]);

    return std::min(minCost[minCost.size() - 1], minCost[minCost.size() - 2]);
  }
};

File: python/0746.py
# Link: https://leetcode.com/problems/min-cost-climbing-stairs/description/
# Runtime: 55 ms
# Memory: 16.22 MB


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        min_cost = cost[:]
        for i in range(2, len(min_cost)):
            min_cost[i] += min(min_cost[i - 1], min_cost[i - 2])

        return min(min_cost[-1], min_cost[-2])
